nanoFramework.Device.Bluetooth/BluetoothLEServer.cs

```diff
@@ -0,0 +1,267 @@
+                        chr.WriteProtectionLevel == GattProtectionLevel.EncryptionAndAuthenticationRequired)
+                    {
+                        secureConnection = true;
+                    }
+
+                    if (chr.ReadProtectionLevel == GattProtectionLevel.AuthenticationRequired ||
+                        chr.ReadProtectionLevel == GattProtectionLevel.EncryptionAndAuthenticationRequired ||
+                        chr.WriteProtectionLevel == GattProtectionLevel.AuthenticationRequired ||
+                        chr.WriteProtectionLevel == GattProtectionLevel.EncryptionAndAuthenticationRequired)
+                    {
+                        authentication = true;
+                    }
+                }
+            }
+
+            if (secureConnection)
+            {
+                _minProtectionLevel = DevicePairingProtectionLevel.Encryption;
+            }
+
+            if (authentication)
+            {
+                _minProtectionLevel = DevicePairingProtectionLevel.EncryptionAndAuthentication;
+            }
+
+            return _minProtectionLevel;
+        }
+
+        #endregion Security
+
+        /// <summary>
+        /// Start Bluetoth stack for server mode
+        /// If already running or in Client mode will give an InvalidOperation Exception.
+        /// </summary>
+        public void Start()
+        {
+            // Check and switch to server mode ( Stack Bluetooth stack )
+            BluetoothNanoDevice.CheckMode(BluetoothNanoDevice.Mode.Server);
+        }
+
+        /// <summary>
+        /// Stop Bluetooth server mode.
+        /// This stops the bluetooth stack but doesn't remove the current Services managed objects.
+        /// To remove these dispose the BluetoothLEServer instance.
+        /// </summary>
+        public void Stop()
+        {
+            // Check in Server mode and change if not
+            BluetoothNanoDevice.CheckMode(BluetoothNanoDevice.Mode.NotRunning);
+        }
+
+        /// <summary>
+        /// Route Bluetooth events
+        /// </summary>
+        /// <param name="btEvent"></param>
+        internal void OnEvent(BluetoothEventSesssion btEvent)
+        {
+            //Debug.WriteLine($"# BluetoothLEServer OnEvent, type:{btEvent.type} status:{btEvent.status}");
+
+            switch (btEvent.type)
+            {
+                case BluetoothEventType.ClientConnected:
+                case BluetoothEventType.ClientDisconnected:
+                    _session.OnEvent(btEvent);
+                    break;
+
+                case BluetoothEventType.ClientSessionChanged:
+                case BluetoothEventType.PassKeyActions:
+                case BluetoothEventType.PassKeyActions_numcmp:
+                case BluetoothEventType.AuthenticationComplete:
+                    Pairing.OnEvent(btEvent);
+                    break;
+
+                default:
+                    break;
+            }
+        }
+
+        /// <summary>
+        /// Dispose
```

Comment:
you can use the inherit doc with the default /// comment

Reply:
Didn't know about that one.